pkgs/darkest-dungeon/default.nix

```diff
@@ -0,0 +1,100 @@
+{
+  lib,
+  requireFile,
+  makeDesktopItem,
+  copyDesktopItems,
+  autoPatchelfHook,
+  makeBinaryWrapper,
+  stdenvNoCC,
+  zip,
+  unzip,
+  SDL2,
+  libcxx,
+  libGL,
+  fmodex,
+  libmx,
+  libX11,
+  libXext,
+  libxcb,
+  libXau,
+  libXdmcp,
+}: let
+  commonDeps = [
+    SDL2
+    libcxx
+    libGL
+    fmodex
+    libmx
+    fmodex
+    libX11
+    libXext
+    libxcb
+    libXau
+    libXdmcp
+  ];
+in
+  stdenvNoCC.mkDerivation rec {
+    pname = "darkest-dungeon";
+    version = "24839_28859";
+
+    # The official installer must be provided by the end user.
+    src = requireFile {
+      name = "darkest_dungeon_${version}.sh";
+      url = "https://www.gog.com/game/darkest_dungeon";
+      sha256 = "2b235b43d8d6ab4b2a4b80b627ce2a9c31c1b4bee16fc6f9dda68c29cb5da99c";
+    };
+
+    nativeBuildInputs =
+      [
+        autoPatchelfHook
+        makeBinaryWrapper
+        unzip
+        zip
+        copyDesktopItems
+      ]
+      ++ commonDeps;
+
+    desktopItems = [
+      (makeDesktopItem {
+        name = pname;
+        desktopName = "Darkest Dungeon";
+        genericName = "Loosing your will to live; the game.";
+        exec = "darkest";
+        #icon = "CCicon";  # The proper icon needs to be set instead.
+        comment = "Darkest Dungeon GoG version.";
+        categories = ["Game"];
+      })
+    ];
+
+    # FIXME: I know this is curserd, but for some reason, it is required.
```

Comment:
We may need more explanations on this. Maybe we can find a solution.

Reply:
This is also not ideal because building the derivation uses the double of space and, by default, Nix has its tmp dir in a tmpfs so it might be a problem on some low end machines.

Reply:
And regarding the problem: GoG linux installers are just shellscripts which also contain a zip archive. So due to the file extension nix does not pick them up as zips and fails to extract them.
I've tried to manually tell it to extract it with `unzip` _(which just works if not in nix)_ but it outputed an error after extracting it...

I need to research it a bit more, but I'm working on it.

Reply:
What about renaming in `preUnpack`?

Reply:
I hadn't thought of that. I'll try that next.
Thanks for the tip.

Reply:
Fixed by commit ff6b8265b738bc12123e474961bcf518f11fefcb. Now the unpack phase is not only less hacky but also more efficient.